dask/backends.py

```diff
@@ -88,7 +90,23 @@ def dispatch(self, backend: str):
                 return self.register_backend(backend, entrypoints[backend].load()())
         else:
             return impl
-        raise ValueError(f"No backend dispatch registered for {backend}")
+
+        # Append additional info for the cudf backend.
+        # Note that the array-backend will already inform
+        # the user to install cupy when that dispatch fails
+        extra = ""
+        if backend == "cudf":
+            try:
+                import dask_cudf
+
+            except ImportError:
+                version = "0.0"
+            else:
+                version = dask_cudf.__version__
+            if parse_version(version) < parse_version("22.12"):
+                extra = " Please try installing dask_cudf>=22.12"
```

Comment:
This works, but it seems off to have logic for the `cudf` backend in the `CreationDispatch` base class. I'm curious if you have thoughts on other approaches. A couple things that come to mind (not necessarily saying these are great ideas):

- Make the name of the backend correspond to the package that provides the backend. This would at least make it so we know the name of the package that needs to be installed without actually having it installed.
- Instead of including the name of the package here, point users to the relevant page in the docs where we could list the known backends and the package that needs to be installed to use them.

Reply:
I share your concern.

I agree that a fundamental problem here is that the entrypoint "cudf" backend is actually defined in `dask_cudf` (rather than `cudf`). With that said, I suspect that we **don't** want to require the backend entrypoint to be defined in a library that has the exact same name as the backend label. I'll admit that this opinion is mostly because I'd like to continue supporting `dask.config.set({"dataframe.backend": "cudf"})` (and not require this to be "dask_cudf"), but it's also because the "pandas", "numpy" and "cupy" entrypoints are all provided by dask itself (and not those "backend" libraries) - Making this requirement a bit confusing.

I suppose it may make the most sense to simply link the docs and allow the docs to elaborate on the need to install `dask_cudf` to use the "cudf" backend. For example:

```python
f"Please make sure your environment provides the necessary `DaskBackendEntrypoint` definition for the {backend} backend. See the backend-dispatching documentation for more information: <link>"
```

(with more information about the required libraries for "pandas", "numpy", "cupy", and "cudf" at the link)

Reply:
> the "pandas", "numpy" and "cupy" entrypoints are all provided by dask itself (and not those "backend" libraries) - Making this requirement a bit confusing.

Fair point. Agreed that this approach seems problematic. 

> I suppose it may make the most sense to simply link the docs and allow the docs to elaborate on the need to install dask_cudf to use the "cudf" backend.

+1. I'd prefer this approach.